### DISCUSSION TITLE
ESP32 C3 SuperMini Infrared binary building

### DISCUSSION THREAD

**Discussion by m4dm4n** [arendst/Tasmota#20239] [Q&A] [answered]
Hi,

Since there is only Tasmota32-CDC binary that is currently compatible with the board, I tried to custom compile Tasmota in Gitpod and enabling the full Infrared support. But for the life of me, I can not find the line where I can either specify C3 board, or more specifically CDC support (which is crucial for succesfull flashing of the board apparently).

Is anyone willing to show me the direction where I'm supposed to be heading? Thank you.

**Comment by barbudor** [comment]
You need to define what is needed for full IR in your user_config_override.h

Then build tasmota32c3cdc with
'platformio run -e tasmota32c3cdc'
Or choose tasmota32c3cdc as the target environment in the bottom line of the editor and then click the checkmark to build

**Reply by barbudor** [reply]
Delete your '.pio' and '.cache' before the build to insure that the IR lib is build properly

**Reply by sfromis** [reply]
While USB-CDC support can be very handy to have a serial console after booting, it is not a condition for being able to flash.

**Reply by m4dm4n** [reply]
@barbudor I deleted .pio and -cache before the build, and the firmware does not work, this message in log:

ESP-ROM:esp32c3-api1-20210207
Build:Feb  7 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0xf (SPI_FAST_FLASH_BOOT)
Saved PC:0x40047ed2
SPIWP:0xee
mode:DIO, clock div:1
load:0x3c150020,len:0x9afe8
load:0x3fc8da00,len:0x2ff0
load:0x40380000,len:0x2010
load:0x42000020,len:0x146df8
Invalid image block, can't boot.
ets_main.c 333 

Or I even can't get any serial communication.

I installed CDC version again via web installer (or sometimes even had to force it via esptool because web install couldn't connect to it), and it works.

I tried even to compile completely clean version, without any changes, only 'platformio run -e tasmota32c3cdc', and the firmware does not work. So, the official firmwares have some extra config changes that I'm not aware of.

In any situation, only official Tasmota32C3CDC.bin works, but I need the full IR support for HVAC control.

**Reply by sfromis** [reply, marked as answer]
To flash a new build, not as an upgrade, you should use the `.factory.bin` variant produced by the build. The `.factory` images includes bootloader, safeboot and partitioning, along with the "regular" binary.

**Reply by m4dm4n** [reply]
@sfromis thank you man, sorry I crashed the party like a complete noob, every day you learn something new. Managed to implement the IR library, USB communication also works, just to let everyone else know in the future.
